kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/test/java/org/kie/server/controller/websocket/management/KieServerMgmtCommandServiceImplTest.java

```diff
@@ -0,0 +1,219 @@
+
+        assertNotNull(response);
+        assertEquals(SUCCESS,
+                     response.getType());
+    }
+
+    @Test
+    public void testCommandSerialization() {
+        final ServerTemplate serverTemplate = new ServerTemplate();
+        serverTemplate.setId("templateId");
+        serverTemplate.setName("templateName");
+        KieServerControllerDescriptorCommand command = new KieServerControllerDescriptorCommand(SpecManagementService.class.getName(),
+                                                                                                "saveServerTemplate",
+                                                                                                serverTemplate);
+
+        final String content = WebSocketUtils.marshal(marshallingFormat.getType(),
+                                                      command);
+
+        LOGGER.info("{} content\n{}", marshallingFormat.getType(), content);
+        KieServerControllerServiceResponse response = service.executeCommand(WebSocketUtils.unmarshal(content,
+                                                                                                      marshallingFormat.getType(),
+                                                                                                      KieServerControllerDescriptorCommand.class));
+
+        assertNotNull(response);
+        assertEquals(response.getMsg(),
+                     SUCCESS,
+                     response.getType());
+        assertNull(response.getResult());
+
+        command = new KieServerControllerDescriptorCommand(SpecManagementService.class.getName(),
+                                                           "getServerTemplate",
+                                                           (Object) "templateId");
+
+        response = service.executeCommand(command);
+
+        assertNotNull(response);
+        assertEquals(SUCCESS,
+                     response.getType());
+        assertNotNull(response.getResult());
+        assertEquals(serverTemplate,
+                     response.getResult());
+
+        String responseContent = WebSocketUtils.marshal(marshallingFormat.getType(),
+                                                        response);
+        response = WebSocketUtils.unmarshal(responseContent,
+                                            marshallingFormat.getType(),
+                                            KieServerControllerServiceResponse.class);
+        assertNotNull(response);
+        assertEquals(SUCCESS,
+                     response.getType());
+        assertNotNull(response.getResult());
+        assertEquals(serverTemplate,
+                     response.getResult());
+    }
+
+    @Test
+    public void testCommandWithDomainArguments() {
+        final ContainerSpec containerSpec = new ContainerSpec();
+        containerSpec.setId("templateId");
+        containerSpec.setContainerName("containerId");
```

Comment:
Container id is a "templateId" and container name is "containerId"?
Use there String constants for container ID and container name.